### PR TITLE
New Resource: aws_waf_regex_match_set

### PR DESCRIPTION
**Pull request by radeksimko**
## Test results

```
TF_ACC=1 go test ./aws -v -run=TestAccAWSWafRegexMatchSet_ -timeout 120m
=== RUN   TestAccAWSWafRegexMatchSet_basic
--- PASS: TestAccAWSWafRegexMatchSet_basic (46.22s)
=== RUN   TestAccAWSWafRegexMatchSet_changePatterns
--- PASS: TestAccAWSWafRegexMatchSet_changePatterns (73.11s)
=== RUN   TestAccAWSWafRegexMatchSet_noPatterns
--- PASS: TestAccAWSWafRegexMatchSet_noPatterns (32.71s)
=== RUN   TestAccAWSWafRegexMatchSet_disappears
--- PASS: TestAccAWSWafRegexMatchSet_disappears (42.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	194.634s
```

### CODE REVIEW

File: aws/resource_aws_waf_regex_match_set.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data": {
+										Type:     schema.TypeString,
+										Optional: true,
+										StateFunc: func(v interface{}) string {
+											value := v.(string)
+											return strings.ToLower(value)
+										},
```
Comment (appilon) [inline aws/resource_aws_waf_regex_match_set.go]: one-liner? I'm not convinced either way it's good style but I'll suggest it and let you decide:
```
return strings.ToLower(v.(string))
```

File: aws/resource_aws_waf_regex_match_set.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		}
+	}
+
+	wr := newWafRetryer(conn, "global")
+	_, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+		req := &waf.DeleteRegexMatchSetInput{
+			ChangeToken:     token,
+			RegexMatchSetId: aws.String(d.Id()),
+		}
+		log.Printf("[INFO] Deleting WAF WAF Regex Match Set: %s", req)
```
Comment (appilon) [inline aws/resource_aws_waf_regex_match_set.go]: nitpick: double `WAF` in message

File: aws/resource_aws_waf_regex_match_set_test.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+			return fmt.Errorf("Error updating WAF Regex Match Set: %s", err)
+		}
+
+		wr := newWafRetryer(conn, "global")
+		_, err = wr.RetryWithToken(func(token *string) (interface{}, error) {
+			req := &waf.DeleteRegexMatchSetInput{
+				ChangeToken:     token,
+				RegexMatchSetId: aws.String(*set.RegexMatchSetId),
+			}
+			log.Printf("[INFO] Deleting WAF WAF Regex Match Set: %s", req)
```
Comment (appilon) [inline aws/resource_aws_waf_regex_match_set_test.go]: nitpick double WAF

File: aws/resource_aws_waf_regex_match_set.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data": {
+										Type:     schema.TypeString,
+										Optional: true,
+										StateFunc: func(v interface{}) string {
+											value := v.(string)
+											return strings.ToLower(value)
```
Comment (appilon) [inline aws/resource_aws_waf_regex_match_set.go]: nitpick, consider one liner? `return strings.ToLower(v.(string))`

File: aws/resource_aws_waf_regex_match_set_test.go [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+		wr := newWafRetryer(conn, "global")
+		_, err := wr.RetryWithToken(func(token *string) (interface{}, error) {
+			req := &waf.UpdateRegexMatchSetInput{
+				ChangeToken:     token,
+				RegexMatchSetId: set.RegexMatchSetId,
+			}
+
+			for _, tuple := range set.RegexMatchTuples {
+				update := &waf.RegexMatchSetUpdate{
```
Comment (appilon) [inline aws/resource_aws_waf_regex_match_set_test.go]: nitpick, consider 1 liner:
```
req.Updates = append(req.Updates, &waf.RegexMatchSetUpdate{
  Action: aws.String("DELETE"),
  RegexMatchTuple: tuple,
})
```

**Comment by radeksimko** [comment]
Thanks for the review, I believe I addressed all your feedback. 😉

**Comment by bflad** [comment]
This has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**